src/raft/recv_install_snapshot.h

```diff
@@ -3,8 +3,47 @@
 #ifndef RECV_INSTALL_SNAPSHOT_H_
 #define RECV_INSTALL_SNAPSHOT_H_
 
+#include <sqlite3.h>
+
 #include "../raft.h"
 
+// Forward declaration.
+struct snapshot_leader_state;
+
+struct snapshot_leader_io {
+	bool (*log_index_found)(raft_index index);
+	int (*async_work)(struct raft_io_async_work *req,
+			  raft_io_async_work_cb cb);
+	void (*async_send_message)(struct raft_io_send *req,
+			struct raft_message *msg,
+			raft_io_send_cb cb);
+	int (*read_page)(uint64_t offset,
+			int64_t page_size,
+			char *page);
+	// read_chunk
+};
+
+struct snapshot_leader_state {
+	struct sm sm;
+	raft_id follower_id;
+	sqlite3 *ht;
+	sqlite3_stmt *ht_stmt;
```

Comment:
Generally I think we want to have all mentions of sqlite3 types and functions on the dqlite side of the boundary.